os/fs/smartfs/smartfs_utils.c

```diff
@@ -966,15 +963,14 @@ int smartfs_finddirentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *d
 							direntry->dsector = readwrite.logsector;
 							direntry->doffset = offset;
 							direntry->dfirst = dirstack[depth];
+
 							if (direntry->name == NULL) {
-								direntry->name = (char *)kmm_malloc(fs->fs_llformat.namesize + 1);
+								direntry->name = (char *)kmm_zalloc(fs->fs_llformat.namesize + 1);
```

Comment:
anyway we have to use direntry->name, instead of filename, then we can allocate it at the start of function.

Reply:
direntry->name will need space to be allocated but not always.
Line - 1103, direntry->name is left as NULL

Reply:
I mean normally we should use it, it is rare case, it can be free in errout.